setup.py

```diff
@@ -235,27 +234,40 @@ def get_build_overrides():
     from numpy.distutils.command.build_clib import build_clib
     from numpy.distutils.command.build_ext import build_ext
 
-    def _is_using_gcc(obj):
-        is_gcc = False
+    def _is_using_old_gcc(obj):
+        is_old_gcc = False
         if obj.compiler.compiler_type == 'unix':
             cc = sysconfig.get_config_var("CC")
             if not cc:
                 cc = ""
             compiler_name = os.path.basename(cc)
-            is_gcc = "gcc" in compiler_name
-        return is_gcc
+            if "gcc" in compiler_name:
+                out = subprocess.run(['gcc', '-dM', '-E', '-'],
```

Comment:
```suggestion
                out = subprocess.run(obj.compiler.compiler + ['-dM', '-E', '-'],
```
seems to be how distutils does it...

Reply:
adopted `obj.compiler.compiler`